File: python/set_one.py
# Set One
import base64

def hex_to_base64(s):
    return str(base64.b64encode(bytes.fromhex(s)))[2:-1]

def fixed_xor(s, buffer):
    return hex(int(s, 16) ^ int(buffer, 16))[2:]


    
if __name__ == "__main__":
    self.main()

File: ruby/lib/set_one.rb
# require "base64"

class SetOne
    def hex_to_base64(digest)
        # tbd: https://stackoverflow.com/questions/9986971/converting-a-hexadecimal-digest-to-base64-in-ruby
        # what magical sorcery is this
        [[digest].pack("H*")].pack("m0")
    end

    def fixed_xor(buffer_one, buffer_two)
        buffer_one + buffer_two
    end
end
